### PR TITLE
validate orchestration config

### PR DESCRIPTION
**Pull request by kangyili**
### What does this PR do?
Validate the orchestration config provided by user

### Motivation

A deployment in one of our internal clusters led to cluster workers entering CLB because a collector was activated while the resource type was not available in that cluster.
When a resource is not available in a cluster, the informer initial synchronization call blocks until the context timeout is hit which leads to CLB because the liveness probe fails due to the configpoller being blocked as well.

### Describe how to test/QA your changes
For example, run cluster agent on a cluster not supporting verticalpodautoscalers, then you should be able to see warning 
```
Unsupported collector: verticalpodautoscalers
```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes-dca/notes/validate-config-b6bd131de42c023e.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+enhancements:
+  - |
+    validate the orchestration config provided by user
```
Comment (drichards-87) [inline releasenotes-dca/notes/validate-config-b6bd131de42c023e.yaml:4]:
```suggestion
    Validate the orchestration config provided by the user.
```

**Review by drichards-87 — APPROVED** [review]
Left you a small suggestion from Docs and approved the PR.

File: releasenotes-dca/notes/validate-config-b6bd131de42c023e.yaml (hunk 1 above)
Comment (kangyili) [reply releasenotes-dca/notes/validate-config-b6bd131de42c023e.yaml]: Thanks ! [2796f30](https://github.com/DataDog/datadog-agent/pull/16073/commits/2796f30a2c7d53f4faca24d42afb2a52e3fa1cfb)

File: pkg/collector/corechecks/cluster/orchestrator/discovery/collector_discovery_crd.go [hunk 2]
```diff
@@ -33,29 +35,25 @@ type DiscoveryCollector struct {
...
-	collector, err := d.DiscoverCRDResource(resource, groupVersion)
+	err := dc.fillCache()
 	if err != nil {
-		return nil, err
+		log.Error("Fail to init discovery collector", err)
 	}
-	return collector, nil
+	return dc
 }
-
```
Comment (JLineaweaver) [inline pkg/collector/corechecks/cluster/orchestrator/discovery/collector_discovery_crd.go:48]: Any risk of us returning the dc if a fillCache fails?

Comment (kangyili) [reply pkg/collector/corechecks/cluster/orchestrator/discovery/collector_discovery_crd.go:48]: If fillCache fails, it means we can't find any available resources from the server. 
 
In this case, we won't find any collector here https://github.com/DataDog/datadog-agent/blob/2796f30a2c7d53f4faca24d42afb2a52e3fa1cfb/pkg/collector/corechecks/cluster/orchestrator/discovery/collector_discovery_crd.go#L113-L121

Therefore, our bundle won't be able to load any collector from the config https://github.com/DataDog/datadog-agent/blob/2796f30a2c7d53f4faca24d42afb2a52e3fa1cfb/pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go#L148-L153 , which I think is expected.

**Review by fisherevans — APPROVED** [review]

**Pull request merged**